### ISSUE TITLE
Add Code of Conduct to BitShares-Core Repository

### ISSUE THREAD

**Issue by ryanRfox** [bitshares/bitshares-core#1184]
**User Story**
As a `BitShares Community Member` we expect our project maintainers and contributors to follow a `Code of Conduct` so that `participation in this project remains a harassment-free experience for everyone`, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, or nationality.

**Additional Context (optional)**
The BitShares project was created to foster an open, innovative and inclusive community to support development of the platform, tools and documentation. To clarify expected behavior within our repositories, we __propose to adopt__ the [Contributor Covenant](http://contributor-covenant.org/). This code of conduct has been adopted by [many other open source communities](http://contributor-covenant.org/adopters/) and feel it expresses our community values well. 

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by pmconrad** [comment]
+1 for Pledge, Standards and Scope.

-1 for Our Responsibilities and Enforcement. This binds resources and may open us up to liabilities.

**Comment by jmjatlanta** [comment]
May be able to cull ideas from http://code-of-merit.org

I can't say I am behind all of what it says, just a resource.

Hat Tip: @xeroc

**Comment by jmjatlanta** [comment]
Another interesting source for ideas. Before his death, the maintainer of ZeroMQ wrote this:

https://rfc.zeromq.org/spec:22/C4/

Again, I do not agree with everything there. But it is a source for ideas.

**Comment by clockworkgr** [comment]
After reading through CoC and its background, I'm not happy about it.

It seems the people behind it are very anti-meritocracy and I don't like it. (see: https://postmeritocracy.org/)

Although I agree with parts of the CoC in principle , I would personally prefer if we weren't associated with any kind of anti-meritocracy community which sounds like political correctness gone mad.

**Comment by cogutvalera** [comment]
IMHO we need some kind of **"Code of Ethics"** not only for developers it can be applied but for everybody, for all users, for traders, and so on ...

https://www.investopedia.com/exam-guide/cfa-level-1/ethics-standards/code-ethics.asp

**Comment by pmconrad** [comment]
> https://rfc.zeromq.org/spec:22/C4/

A lot of good stuff, but more about the technical process and less about conduct.

> I would personally prefer if we weren't associated with any kind of anti-meritocracy community

Agree in principle, however, this is not quite the typical community project. Since the core team is acting in its role as "employees" of the blockchain, merit does not play a significant role here.

> https://www.investopedia.com/exam-guide/cfa-level-1/ethics-standards/code-ethics.asp

That's about financial stuff, only applicable here in a *very* broad sense.

**Comment by clockworkgr** [comment]
> > I would personally prefer if we weren't associated with any kind of anti-meritocracy community
> 
> Agree in principle, however, this is not quite the typical community project. Since the core team is acting in its role as "employees" of the blockchain, merit does not play a significant role here.
> 

I'd agree if the team was "fixed" but since the team/repo accepts and interacts with 3rd party contributors , merit plays (or at least should play) a significant role in assigning tasks/bounties.

**Comment by pmconrad** [comment]
Hm, then we disagree. :-)
Merit should play only a very small role in assigning tasks (contributor should have demonstrated competence before being assigned complicated tasks), and no role in paying bounties (result counts).

**Comment by clockworkgr** [comment]
Language barrier... 
a) I don't mean merit plays a role in payment. I meant bounties as in tasks offered to the community (not their effort estimate)
b) What you say is exactly what I mean by merit. (merit: the quality of being particularly good or worthy ...so competence)

The people behind the CoC are proponents of quality of code and intelligence  not being as important as communication/interpersonal/social skill etc which I disagree with.

What's more and this is actually more important, the Code is written in a way to ENFORCE a certain conduct instead of CONDEMNING one.

i.e. It says.. here is what you MUST do so you are allowed to stay. Instead of this is what you CAN'T do or you'll get kicked out (as a Code of Conduct should)

It's a subtle yet important difference.

**Comment by clockworkgr** [comment]
For reference: https://linustechtips.com/main/topic/974038-why-the-linux-coc-is-bad/

**Comment by pmconrad** [comment]
Article (in german): https://www.heise.de/ix/meldung/Code-of-Conduct-fuer-Linux-Mit-Buerokratie-zum-Duckmaeusertum-4176439.html
tl;dr: CoC aims for political control and is used as a means to silence different opinions

**Comment by clockworkgr** [comment]
@pmconrad that's been my impression too

Disclaimer.: I do wanna see a code of conduct...just not that one

**Comment by pmconrad** [comment]
The LKML guys have resolved their CoC discussion by removing the second paragraph of the "Enforcement" section and adding a document that explains how the CoC is to be interpreted. Apart from that, their idea is to use this for now, and adapt if it turns out to create more problems than it solves.

* https://www.kernel.org/doc/html/latest/process/code-of-conduct.html
* https://www.kernel.org/doc/html/latest/process/code-of-conduct-interpretation.html

IMO that's a reasonable approach. We could use both as a template, although the latter requires some effort to adapt.

If we choose to take that route, we should coordinate with the other repo maintainers in the bitshares github project. Best use one set of documents for all.
